frontend/views/containers/settings/UserProfile.vue

```diff
@@ -170,43 +152,47 @@ export default {
       sbp('okTurtles.events/emit', OPEN_MODAL, mode)
       return false
     },
-    async save () {
-      if (this.ephemeral.newPicture) {
-        try {
-          this.form.picture = await imageUpload(this.form.picture)
-          this.ephemeral.newPicture = false
-        } catch (error) {
-          console.error(error)
-          this.ephemeral.errorMsg = L('Failed to upload user picture')
-          return false
+    async fileChange (fileList) {
+      if (!fileList.length) return
+      const fileReceived = fileList[0]
+      let picture
+
+      this.$refs.picture.setFromBlob(fileReceived)
+
+      try {
+        picture = await imageUpload(fileReceived)
+      } catch (error) {
+        console.error(error)
+        this.$refs.pictureMsg.danger(`${L('Failed to upload avatar. Please, try again.')} ${error.message}`)
+        return false
+      }
+
+      await this.setAttributes({ picture }, 'pictureMsg', L('Picture updated!'))
+    },
+    async saveProfile () {
+      this.$refs.formMsg.clean()
+      const attrs = {}
+
+      for (const key in this.form) {
+        if (this.form[key] !== this.attributes[key]) {
+          attrs[key] = this.form[key]
         }
       }
 
+      await this.setAttributes(attrs, 'formMsg', L('Your changes were saved!'))
+    },
+    async setAttributes (attrs, refFeedback, successMsg) {
       try {
-        this.ephemeral.profileSaved = false
-        const attrs = {}
-        for (const key in this.form) {
-          if (this.form[key] !== this.attributes[key]) {
-            attrs[key] = this.form[key]
-          }
-        }
         const attributes = await sbp('gi.contracts/identity/setAttributes/create',
           attrs,
           this.$store.state.loggedIn.identityContractID
         )
         await sbp('backend/publishLogEntry', attributes)
-        this.ephemeral.profileSaved = true
-      } catch (ex) {
-        // TODO: handle better
-        console.error(ex)
-        this.ephemeral.errorMsg = L('Failed to Save Profile')
+        this.$refs[refFeedback].success(successMsg)
+      } catch (error) {
+        console.error(error)
+        this.$refs[refFeedback].danger(`${L('Something went wrong, please try again.')} ${error.message}`)
```

Comment:
`${L('Something went wrong, please try again.')} ${error.message}` => `L('Error saving profile: {errorMsg}', { errorMsg: error.message })`

Reply:
cc @mmbotelho

Reply:
I don't understand what you need from me here - can you clarify?

Reply:
Ah I understand now. I have the [same question](https://github.com/okTurtles/group-income-simple/pull/779#discussion_r350760545) - can you please give me examples of what this would be? And again, do the users need to know the exact error or can we keep it simple?